packages/design-system/package.json

```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@repo/design-system",
+  "version": "0.0.0",
+  "exports": {
+    ".": {
+      "import": "./dist/index.mjs",
+      "types": "./index.ts"
```

Comment:
It looks like you've added this package as a "buildable package" and not a "source package" (or whatever they call it these days). Basically meaning you build the design system and the consumers in this repo such as the docs etc install it as a package and uses the built/dist output.

This is fine and what I would have expected personally, however because you're doing this I wouldn't reference the `index.ts` file as your types in the package json file. Instead, I suggest you install the DTS plugin and use that instead: https://github.com/qmhc/unplugin-dts

This will generate types in the dist output so when it's published somewhere, consumers have access to it also. Then you'll update the package.json fle to point to the newly generates `./dist/src/index.d.ts` (guess at dist path).